pennylane/devices/default_qubit.py

```diff
@@ -31,8 +31,27 @@
 
 # tolerance for numerical errors
 tolerance = 1e-10
+SQRT2INV = 1 / np.sqrt(2)
+TPHASE = np.exp(1j * np.pi / 4)
 
 
+def _get_slice(index, axis, num_axes):
+    """Allows slicing along an arbitrary axis of an array or tensor.
```

Comment:
Could be worth adding an example, would also help understanding the exact role of each argument.

Reply:
Also, in the quantum sense, is it fair to say that this is a `_get_subsystem` function? If so and didn't miss anything, might be worth a renaming, just because that it might make it more intuitive when reading how the operations are being applied

Reply:
Thanks, have added an example: 9efdbd5c71399cb913e41f7459ac4a0cf3b7c0f9.

Reply:
> is it fair to say that this is a _get_subsystem function?

No, it's slightly different. In a quantum sense, suppose we have a three qubit system with state tensor having shape `(2, 2, 2)`. Then `_get_slice(1, 1, 3)` gives us all of the amplitudes that correspond to having a |1> state in qubit 1. Likewise, `_get_slice(1, 2, 3)` would be all the amplitudes with a |1> state in qubit 2, while `_get_slice(0, 1, 3)` would be all the amplitudes with a |0> state in qubit 1. So really what `_get_slice()` allows you to do is to slice into the state so that your selecting |0> or |1> for a given qubit. This is relevant when, for example, we want to do control or when we want to apply a phase on the |1> state.

Unfortunately, I'm not sure of the best name :laughing: